smolyay/samples.py

```diff
@@ -0,0 +1,409 @@
+import abc
+import numpy
+
+
+class UnidimensionalPointSet(abc.ABC):
+    """Set of unidimensional points
+
+    A set of unique unidimensional points within a domain.
+    """
+
+    def __init__(self, domain):
+        self._points = None
+        self._domain = numpy.array(domain)
+        self._valid_cache = False
+
+    @property
```

Comment:
Looking at the code, I think it would be OK to add a setter for the domain that invalidates the point cache. We can worry upstream (in the multivariable point sets or the surrogates) about using these in an immutable way.

Reply:
Would we really want to recalculate all the all after a domain change? We could just scale the existing points to the new domain.

Reply:
This implementation is the quickest to code (although maybe inefficient), so I would do it this way and then you can always optimize it later if you find it’s an issue with profiling.

Reply:
okay